File: .NET Core/Api/ApiEndpoint.cs
namespace Contidio.Sdk.Api
{
    public sealed class ApiEndpoint
    {
        internal string Endpoint { private set; get; }
        internal int Port { private set; get; }
        internal string Protocol { private set; get; }
        internal bool IsFrontend { private set; get; }

        internal ApiEndpoint(bool isHttps, string endpoint, int port, bool isFrontend)
        {
            Endpoint = endpoint;
            Port = port;

            if (isHttps)
                Protocol = "https";
            else
                Protocol = "http";

            IsFrontend = isFrontend;
        }

        public string GetApiEndpointUrl(
            ApiEnvironment environment,
            bool apiManagement)
        {
            if (apiManagement)
            {
                switch (environment)
                {
                    case ApiEnvironment.DEVELOPMENT:
                        return
                            Protocol + "://" + Endpoint + "-dev.contidio.com:" + Port;
                    case ApiEnvironment.STAGING:
                        return
                            Protocol + "://" + Endpoint + "-staging.contidio.com:" + Port;
                    case ApiEnvironment.DEMO:
                        return
                            Protocol + "://" + Endpoint + "-demo.contidio.com";                                            
                    default:
                        return
                            Protocol + "://" + Endpoint + ".contidio.com";
                }
            }
            else
            {
                switch (environment)
                {
                    case ApiEnvironment.DEVELOPMENT:
                        return
                            Protocol + "://" + Endpoint + "-dev.contidio.com:" + Port;
                    case ApiEnvironment.STAGING:
                        return
                            Protocol + "://" + Endpoint + "-staging.contidio.com:" + Port;
                    case ApiEnvironment.DEMO:
                        return
                            Protocol + "://" + Endpoint + "-demo.contidio.com";
                    default:
                        return
                            Protocol + "://" + Endpoint + ".contidio.com";
                }
            }
        }        
    }
}

File: Java/src/main/java/com/contidio/sdk/api/ApiEndpoint.java
package com.contidio.sdk.api;

public final class ApiEndpoint {
	private String endpoint;
	private int port;	
	
	private String protocol;
	
	private boolean isFrontend;
	
	ApiEndpoint(boolean isHttps, String endpoint, int port, boolean isFrontend) {
		this.endpoint = endpoint;
		this.port = port;
		
		if (isHttps)
			protocol = "https";
		else
			protocol = "http";
		
		this.isFrontend = isFrontend;
	}
	
	String getEndpoint() {
		return endpoint;
	}
	
	int getPort() {
		return port;
	}
	
	boolean isFrontend() {
		return isFrontend;
	}

	public String getApiEndpointUrl(
			ApiEnvironment environment, 
			boolean apiManagement) 
	{
		if (apiManagement) {
			switch (environment) {
				case DEVELOPMENT:
					return
						protocol + "://" + getEndpoint() + "-dev.contidio.com:" + getPort();					
				case STAGING:
					return
						protocol + "://" + getEndpoint() + "-staging.contidio.com:" + getPort();
				case DEMO:
					return
						protocol + "://" + getEndpoint() + "-demo.contidio.com";
				default:
					return
						protocol + "://" + getEndpoint() + ".contidio.com";
			}
		} else {
			switch (environment) {
				case DEVELOPMENT:
					return
						protocol + "://" + getEndpoint() + "-dev.contidio.com:" + getPort();
				case STAGING:
					return
						protocol + "://" + getEndpoint() + "-staging.contidio.com:" + getPort();					
				case DEMO:
					return
						protocol + "://" + getEndpoint() + "-demo.contidio.com";
				default:
					return
						protocol + "://" + getEndpoint() + ".contidio.com";
			}
		}
	}
}
